rotkehlchen/exchanges/htx.py

```diff
@@ -0,0 +1,183 @@
+        new_params['SignatureMethod'] = 'HmacSHA256'
+        new_params['Timestamp'] = datetime.datetime.now(datetime.UTC).strftime('%Y-%m-%dT%H:%M:%S')
+
+        host = urllib.parse.urlparse(url).hostname
+        path = urllib.parse.urlparse(url).path
+        param_str = '&'.join(  # params need to be sorted to be correctly validated
+            [
+                str(k) + '=' + urllib.parse.quote_plus(str(v))  # need to use the quoted string since cursors have `=` and it breaks signatures  # noqa: E501
+                for k, v in sorted(new_params.items())
+                if v is not None
+            ],
+        )
+        # api describes that strings should be joined using '\n'
+        payload = f'GET\n{host}\n{path}\n{param_str}'
+        dig = hmac.new(
+            key=self.secret,
+            msg=payload.encode('utf-8'),
+            digestmod=hashlib.sha256,
+        ).digest()
+        new_params['Signature'] = base64.b64encode(dig).decode()
+        return new_params
+
+    def _query(self, absolute_path: str) -> Any:
+        url = f'{self.base_url}{absolute_path}'
+        signed_payload = self._sign_request(url=url, params={})
+        try:
+            response = self.session.get(
+                url=url,
+                params=signed_payload,
+                timeout=CachedSettings().get_timeout_tuple(),
+            )
+            res_body = response.json()
+        except requests.RequestException as e:
+            raise RemoteError(f'Failed to query HTX api due to {e}') from e
+
+        if res_body.get('status') != 'ok':
+            raise RemoteError(f'Got an error response from HTX. {response}')
+
+        if (body := res_body.get('data')) is None:
+            raise RemoteError(f'Requested HTX endpoint does not have a data attribute. {response}')
+
+        return body
+
+    def get_accounts(self) -> list[dict[str, Any]]:
+        """Query the accounts to which the api key has access to"""
+        return self._query(absolute_path='/v1/account/accounts')
+
+    def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+        """Query balances for the accounts linked to the api key"""
+        returned_balances: dict[AssetWithOracles, Balance] = {}
+        for account in self.get_accounts():
+            account_id = account['id']
+            path = f'/v1/account/accounts/{account_id}/balance'
+            try:
+                data = self._query(path)
+            except RemoteError as e:
+                error_prefix = 'Failed to query HTX'
+                log.error(f'{error_prefix} balances due to {e}')
+                return returned_balances, f'{error_prefix} due to a remote error. Check logs for more details'  # noqa: E501
+
+            if (account_balance_type := data['type']) is None:
+                log.error(f'Response for balances does not contain the type key {data}. Skipping')
+                continue
+            if account_balance_type not in self.valid_account_balances_types:
```

Comment:
Would it make sense to log deug here for something being ignored as this could help us in debugging? Can also then get rid of the comment as the log would explain it